uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequest.php

```diff
@@ -25,10 +66,59 @@ public function __construct(LeaveBalanceChangeService $leaveBalanceChangeService
    * @return \CRM_HRLeaveAndAbsences_BAO_LeaveRequest|NULL
    */
   public function create($params, $validate = true) {
-    $leaveRequest = LeaveRequest::create($params, $validate);
-    $this->leaveBalanceChangeService->createForLeaveRequest($leaveRequest);
+    if ($validate) {
+      LeaveRequest::validateParams($params);
+    }
 
-    return $leaveRequest;
+    if(!$this->canCreateAndUpdateLeaveRequestFor($params['contact_id'])) {
+      throw new RuntimeException('You are not allowed to create or update a leave request for this employee');
+    }
+
+    if(!empty($params['id'])) {
+      return $this->update($params);
+    }
+
+    if(!$this->isValidStatusTransition('', $params['status_id'], $params['contact_id'])) {
+      throw new RuntimeException("You can't create a Leave Request with this status");
```

Comment:
There's a word with a single quote in the sentence already. That's why I used double quotes.